tests/e2e/specs/backend/product-query/product-filters.test.ts

```diff
@@ -0,0 +1,273 @@
+/**
+ * External dependencies
+ */
+import { canvas, setPostContent, insertBlock } from '@wordpress/e2e-test-utils';
+import {
+	visitBlockPage,
+	saveOrPublish,
+	selectBlockByName,
+	findToolsPanelWithTitle,
+	getFixtureProductsData,
+	getFormElementIdByLabel,
+	shopper,
+	getToggleIdByLabel,
+} from '@woocommerce/blocks-test-utils';
+import { ElementHandle } from 'puppeteer';
+import { setCheckbox, unsetCheckbox } from '@woocommerce/e2e-utils';
+
+/**
+ * Internal dependencies
+ */
+import {
+	GUTENBERG_EDITOR_CONTEXT,
+	describeOrSkip,
+	waitForCanvas,
+	openBlockEditorSettings,
+} from '../../../utils';
+
+const block = {
+	name: 'Product Query',
+	slug: 'core/query',
+	class: '.wp-block-query',
+};
+
+/**
+ * Selectors used for interacting with the block in the editor. These selectors
+ * can be changed upstream in Gutenberg, so we scope them here for
+ * maintainability.
+ *
+ * There are also some labels that are used repeatedly, but we don't scope them
+ * in favor of readability. Unlike selectors, those label are visible to end
+ * users, so it's easier to understand what's going on if we don't scope them.
+ * Those labels can get upated in the future, but the tests will fail and we'll
+ * know to update them.
+ */
+const SELECTORS = {
+	productFiltersDropdownButton: {
+		default:
+			'.components-tools-panel-header .components-dropdown-menu button[aria-expanded="false"]',
+		expanded:
+			'.components-tools-panel-header .components-dropdown-menu button[aria-expanded="true"]',
+	},
+	productFiltersDropdown:
+		'.components-dropdown-menu__menu[aria-label="Product filters options"]',
+	productFiltersDropdownItem: '.components-menu-item__button',
+	editorPreview: {
+		productsGrid: 'ul.wp-block-post-template',
+		productsGridItem:
+			'ul.wp-block-post-template > li.block-editor-block-preview__live-content',
+	},
+	productsGrid: `${ block.class } ul.wp-block-post-template`,
+	productsGridItem: `${ block.class } ul.wp-block-post-template > li.product`,
+	formTokenFieldLabel: '.components-form-token-field__label',
+	tokenRemoveButton: '.components-form-token-field__remove-token',
+};
+
+const toggleProductFilter = async ( filterName: string ) => {
+	const productFiltersPanel = await findToolsPanelWithTitle(
+		'Product filters'
```

Comment:
Ouch, this is already going to break soon. According to this design spec (pdnLyh-2By-p2) this label is going to be changed to `Advanced filters`. Implemented in this PR: https://github.com/woocommerce/woocommerce-blocks/pull/7687 (still in review).

I wonder if we should:

a. Deal with it and recognize that we are in a phase in which things are going to change a lot. But will eventually settle.
b. Just give this tools panel a specific class or `test-id` to make sure that whatever the label change, this will keep working.

Reply:
I'd go with a). Updating tests is a normal process. https://github.com/woocommerce/woocommerce-blocks/pull/7687 should update the test to reflect the new label change.